app/src/main/java/com/ldlda/chesscom_stats/utils/cache/TimedCache.kt

```diff
@@ -7,24 +7,23 @@ import java.util.concurrent.ConcurrentHashMap
  * Thread-safe for concurrent reads/writes (ConcurrentHashMap).
  */
 class TimedCache<T>(private val ttlMillis: Long) {
-    private data class Entry<T>(val value: T, val expiresAt: Long)
-
     private val map = ConcurrentHashMap<String, Entry<T>>()
 
     fun get(key: String): T? {
         val e = map[key] ?: return null
         val now = System.currentTimeMillis()
-        return if (now < e.expiresAt) e.value else {
+        return if (now - e.time <= ttlMillis) e.value else {
             // lazy eviction
             map.remove(key, e)
             null
         }
     }
 
     fun put(key: String, value: T) {
-        val expires = System.currentTimeMillis() + ttlMillis
-        map[key] = Entry(value, expires)
+        map[key] = Entry(value, System.currentTimeMillis())
     }
 
     fun clear() = map.clear()
-}
\ No newline at end of file
+}
+
+data class Entry<T>(val value: T, val time: Long)
```

Comment:
This should be inside the class
```suggestion

    data class Entry<T>(val value: T, val time: Long)
}
```